integration-tests/test_client_options.py

```diff
@@ -17,3 +19,50 @@ def test_set_ansible_host_info(insights_client):
     ret = insights_client.run("--ansible-host=foo.example.com", check=False)
     assert "Could not update Ansible hostname" not in ret.stdout
     assert ret.returncode == 0
+
+
+def test_group(insights_client, tmp_path):
+    """Test --group option
+    This test generates an archive in offline mode and verifies if the group
+    specified in --group was created and packed
+    """
+
+    group_name = "testing-group"
+    archive_name = "archive.tar.gz"
+    archive_location = tmp_path / archive_name
+    tag_file = None
+
+    # Running insights-client in offline mode to generate archive
+    insights_client.run(
+        "--offline", f"--group={group_name}", f"--output-file={archive_location}"
+    )
+
+    with tarfile.open(archive_location, "r") as tar:
+        dir_name = tar.getnames()[0]
+        for m in tar.getmembers():
+            if m.name == f"{dir_name}/tags.json":
+                tag_file = tar.extractfile(m)
+                tag_file_content = json.loads(tag_file.read())
```

Comment:
`read()` is not needed here, json will do it on its own if you use `load()` instead of `loads()`